### PR TITLE
add ember-data tests

### PR DESCRIPTION
**Pull request by NullVoxPopuli**

### CODE REVIEW

**Comment by pzuraq** [comment]
We should be adding different `ember-data` versions to the try-config as well

**Comment by pzuraq** [comment]
@NullVoxPopuli we'll also need to update the tests, because native getters did not work at all in 2.18. That's how this slipped under the radar, we weren't testing that codepath because we were always using `get`.

**Comment by runspired** [comment]
We don't need to support ember-data 2.18 as we no longer support it ourselves.

**Comment by pzuraq** [comment]
Sure, the tests can be updated to not run on 2.18. This polyfill will continue to support 2.18, since it’s the last version of v2 and will help users to update.

File: tests/unit/data-test.js [hunk 1]
```diff
@@ -1,7 +1,17 @@
...
-import { getContext } from "@ember/test-helpers";
-import { setupApplicationTest } from "ember-qunit";
+import { module, test } from 'qunit';
+import DS from 'ember-data';
+import { getContext } from '@ember/test-helpers';
+import { setupApplicationTest } from 'ember-qunit';
+import { get as emberGet } from '@ember/object';
+import { gte } from 'ember-compatibility-helpers';
+
+function get(obj, key) {
```
Comment (NullVoxPopuli) [inline tests/unit/data-test.js:9]: noice

File: tests/unit/data-test.js [hunk 2]
```diff
@@ -12,50 +22,48 @@ function register(registeredName, object) {
...
+      .toArray()
+      .map(post => get(post, 'name'))
+      .join();
 
-    assert.equal(
-      names,
-      expectedNames,
-      "The correct records are in the array"
-    );
+    assert.equal(names, expectedNames, 'The correct records are in the array');
```
Comment (NullVoxPopuli) [inline tests/unit/data-test.js:50]: different editor settings?

:man_shrugging:

Comment (pzuraq) [reply tests/unit/data-test.js:50]: Just Prettier doing its thing

File: tests/unit/data-test.js [hunk 3]
```diff
@@ -23,8 +24,8 @@ function register(registeredName, object) {
 
 function stubAdapter() {
   let store = getService('store');
-  store.createRecord('post', { name: name1 });
-  store.createRecord('post', { name: name2 });
```
Comment (runspired) [inline tests/unit/data-test.js]: ah good ole' 2.18 async

**Pull request merged**